### PR TITLE
RtlTimeEstimator: Consider minimum ground speed

### PR DESCRIPTION
**Pull request by mbjd**
### Solved Problem

This fixes an issue seen when the wind is faster than the trim airspeed, but not faster than the max airspeed.

In that case the RTL time estimator currently assumes that we always fly at trim airspeed and are thus unable to cover ground in the upwind direction. The result is a very large RTL time estimate, resulting in RTL if configured by `COM_FLTT_LOW_ACT`. However, RTL may actually be possible in time, because we increase the airspeed to satisfy `FW_GND_SPD_MIN`. 

### Solution 

By considering the `FW_GND_SPD_MIN` parameter, we correct this wrong assumption. The RTL remaining time estimate now becomes realistic, if `windspeed <= cas2tas(FW_AIRSPD_MAX) - FW_GND_SPD_MIN` (i.e. when we are able to satisfy the minimum ground speed). 

### Caveat

While this PR is a strict improvement in the given regime (changing the time estimate from very pessimistic to realistic), it makes the remaining time estimate _optimistic_ if we are not able to reach the min ground speed anymore. The RTL time is underestimated and we possibly run out of battery before reaching home.

### Changelog Entry
For release notes:
```
RTL Time Estimate: Fix time overestimate for high wind speeds
```

### Alternatives
 - Explicitly take into account the max (true) airspeed as well
 - Consider the current TECS airspeed setpoint instead? Would automatically handle min ground / max airspeed both correctly (but only in current direction...), but could lead to additional fluctuations in remaining time estimate. 

Both of these add further complexity for a rather extreme edge case (to be entirely correct we need to re-do the CAS TAS conversion...), but they would address the main drawback of an overly optimistic RTL time estimate in strong headwind. 

### Test coverage

Sim sanity check.

 - Rebase this PR on #26467 to get wind in SIH
 - Fly `sihsim_standard_vtol`, default params, particularly:
```
pxh> param show FW_AIRSPD_*
Symbols: x = used, + = saved, * = unsaved
x   FW_AIRSPD_FLP_SC [545,706] : 1.0000
x   FW_AIRSPD_MAX [546,707] : 12.0000
x   FW_AIRSPD_MIN [547,708] : 7.0000
x   FW_AIRSPD_STALL [548,709] : 7.0000
x   FW_AIRSPD_TRIM [549,710] : 8.0000
```
 - Loiter a couple 100 meters south of the home position
 - Slowly increase the wind speed in south direction

Before PR: RTL time estimate goes to infinity exactly when wind estimate surpasses trim airspeed
image

After PR: RTL time estimate never goes to infinity (not even when wind surpasses max airspeed)
image

### CODE REVIEW

**Review by mahima-yoga — APPROVED** [review]
I think it makes sense and it's an improvement compared to the current logic. 

If we are concerned about high wind conditions (where we cannot achieve minimum ground speed), we could check whether the minimum ground speed is achievable given the wind / max airspeed.

File: src/lib/rtl/rtl_time_estimator.cpp [hunk 1]
```diff
@@ -142,7 +143,19 @@ float RtlTimeEstimator::getCruiseGroundSpeed(const matrix::Vector2f &direction_n
...
 
-		cruise_speed = ground_speed;
+		// Assume that minimum ground speed is always satisfied. If
+		//   windspeed < cas2tas(FW_AIRSPD_MAX) - FW_GND_SPD_MIN
+		// the assumption always holds. Otherwise it breaks down when
+		// flying upwind, and the RTL time estimate is optimistic.
+
+		float fw_gnd_spd_min;
+
+		if (_param_fw_gnd_spd_min == PARAM_INVALID || param_get(_param_fw_gnd_spd_min, &fw_gnd_spd_min) != PX4_OK) {
```
Comment (sfuhrer) [inline src/lib/rtl/rtl_time_estimator.cpp]: I would in that case init the fw_gnd_spd_min to 5 and still use it to constrain, instead of setting the cruise speed to the groundspeed without constraining.

Comment (mbjd) [reply src/lib/rtl/rtl_time_estimator.cpp]: done https://github.com/PX4/PX4-Autopilot/pull/26405/commits/35666e665836f751ced83982cb8162e76f669df6

**Comment by mbjd** [comment]
Build failure [unrelated](https://github.com/PX4/PX4-Autopilot/pull/26478), builds locally:

```
balduin@balduin-thinkpad:~/PX4-Autopilot$ make
[4/4] Linking CXX executable bin/px4
balduin@balduin-thinkpad:~/PX4-Autopilot$ git show
commit 35666e665836f751ced83982cb8162e76f669df6 (HEAD -> pr-consider-min-gnd-spd-in-rtl-time, upstream/pr-consider-min-gnd-spd-in-rtl-time)
Author: Balduin <balduin@auterion.com>
Date:   Fri Feb 13 13:25:31 2026 +0100

    RtlTimeEstimator: assume min ground speed of 5 if param unavailable
```

**Review by sfuhrer — APPROVED** [review]

File: src/lib/rtl/rtl_time_estimator.cpp [hunk 2]
```diff
@@ -142,7 +143,18 @@ float RtlTimeEstimator::getCruiseGroundSpeed(const matrix::Vector2f &direction_n
 		const float ground_speed = sqrtf(cruise_speed * cruise_speed - wind_across_dir * wind_across_dir) + fminf(
 						   0.f, wind_along_dir);
 
-		cruise_speed = ground_speed;
+		// Assume that minimum ground speed is always satisfied. If
+		//   windspeed < cas2tas(FW_AIRSPD_MAX) - FW_GND_SPD_MIN
+		// the assumption always holds. Otherwise it breaks down when
+		// flying upwind, and the RTL time estimate is optimistic.
+
+		float fw_gnd_spd_min = 5.0f;
```
Comment (MaEtUgR) [inline src/lib/rtl/rtl_time_estimator.cpp:151]: Just a general comment: I try to initialize parameter values in a way that would make it very obvious if the parameter loading failed otherwise you might have silent failures not considering the configuration.

Comment (sfuhrer) [reply src/lib/rtl/rtl_time_estimator.cpp:151]: In that case it's better to assume 5 than not doing anything with the min ground speed.
